nightfall/api.py

```diff
@@ -74,38 +80,17 @@ def scan_text(self, text: str, detection_rule_uuids: list = None, detection_rule
             raise NightfallUserError("Need to supply detection rule ids list or detection rules dict with \
                 key 'detection_rule_uuids' or 'detection_rules' respectively", 40001)
 
+        config = {}
         if detection_rule_uuids:
-            return self._handle_detection_rule_uuids_v3(text, detection_rule_uuids)
+            config["detectionRuleUUIDs"] = detection_rule_uuids
         if detection_rules:
-            return self._handle_detection_rules_v3(text, detection_rules)
-
-    def _handle_detection_rule_uuids_v3(self, text, detection_rule_uuids):
-        text_chunked = self._chunk_text(text)
-        all_responses = []
-        for payload in text_chunked:
-            request_body = {
-                "payload": payload,
-                "config": {
-                    "detectionRuleUUIDs": detection_rule_uuids
-                }
-            }
-            response = self._scan_text_v3(request_body)
-            all_responses.extend(response.json()['findings'])
-        return all_responses
-
-    def _handle_detection_rules_v3(self, text, detection_rules):
-        text_chunked = self._chunk_text(text)
-        all_responses = []
-        for payload in text_chunked:
-            request_body = {
-                "payload": payload,
-                "config": {
-                    "detectionRules": detection_rules
-                }
-            }
-            response = self._scan_text_v3(request_body)
-            all_responses.extend(response.json()['findings'])
-        return all_responses
+            config["detectionRules"] = [d.as_dict() for d in detection_rules]
+        request_body = {
+            "payload": text,
+            "config": config
+        }
+        response = self._scan_text_v3(request_body)
+        return response.json()['findings']
```

Comment:
there is now a second key returned https://github.com/nightfallai/nightfall-java-sdk/blob/main/src/main/java/ai/nightfall/scan/model/ScanTextResponse.java#L13-L17